packages/boot/src/booters/controller.booter.ts

```diff
@@ -0,0 +1,125 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/boot
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {CoreBindings, Application, Booter, BootOptions} from '@loopback/core';
+import {inject} from '@loopback/context';
+import {BootBindings} from '../keys';
+import {relative} from 'path';
+import {
+  discoverFiles,
+  discoverFilesWithGlob,
+  loadClassesFromFiles,
+} from './booter-utils';
+
+/**
+ * ControllerBooter is a class that implements the Booter inferface. The purpose
+ * of this booter is to allow for convention based booting of `Controller`
+ * artifacts for LoopBack 4 Applications.
+ *
+ * It supports the following boot phases: config, discover, boot
+ */
+export class ControllerBooter implements Booter {
+  options: ControllerOptions;
+  projectRoot: string;
+  dirs: string[];
+  extensions: string[];
+  discovered: string[];
+
+  /**
+   *
+   * @param app Application instance
+   * @param bootConfig Config options for boot
```

Comment:
Merge the TSDoc on the class and constructor together, please (they serve the same purpose so it keeps things clearer). I don't have a preference for which line they're on, as long as it's just one of them, and is consistent with the others.